pennylane_lightning/src/algorithms/tests/Test_AdjointJacobian.cpp

```diff
@@ -24,13 +26,43 @@ using namespace Pennylane::LightningQubit::Observables;
 } // namespace
 /// @endcond
 
+#elif _ENABLE_PLKOKKOS == 1
+constexpr bool BACKEND_FOUND = true;
+constexpr bool LKOKKOS_FOUND = true;
+
+#include "AdjointJacobianKokkos.hpp"
+#include "ObservablesKokkos.hpp"
+#include "TestHelpersStateVectors.hpp" // TestStateVectorBackends, StateVectorToName
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::LightningKokkos::Util;
+using namespace Pennylane::LightningKokkos::Algorithms;
+using namespace Pennylane::LightningKokkos::Observables;
+} // namespace
+  /// @endcond
+
 #else
 constexpr bool BACKEND_FOUND = false;
+constexpr bool LKOKKOS_FOUND = false;
 using TestStateVectorBackends = Pennylane::Util::TypeList<void>;
 
 template <class StateVector> struct StateVectorToName {};
 #endif
 
+template <class StateVectorT>
+JacobianData<StateVectorT> initJacobianData(
+    size_t num_params, StateVectorT &psi,
+    const std::vector<std::shared_ptr<Observable<StateVectorT>>> &obs,
+    const OpsData<StateVectorT> &ops, const std::vector<size_t> &tp) {
+    if constexpr (LKOKKOS_FOUND) {
+        return {num_params, psi.getLength(), psi.getData().data(), obs, ops,
+                tp};
+    } else {
+        return {num_params, psi.getLength(), psi.getData(), obs, ops, tp};
+    }
+}
+
```

Comment:
100% agree with the goal. Since `getData` pops 158 times in `lightning_kokkos`, I thought it was safer to first get some tests working, and hence this patching and workarounds. I'll give it a shot now.